### PR TITLE
[Poromechanics] Fixing arc length strategy

### PR DESCRIPTION
**Pull request by ipouplana**
Summary of problems addressed:

- The number of iterations were not properly computed. Important for the estimation of the arc length radius.
- We were repeatedly calling the Initialize() method when dofs were reformed. That was problematic because important member variables for the arc length were deleted.

### CODE REVIEW

**Comment by ipouplana** [comment]
@loumalouomega if you are ok with these changes, could you approve it? ^_^'
I just wanted you to know about it, in case you inherited any of these problems...

File: applications/PoromechanicsApplication/custom_strategies/strategies/poromechanics_newton_raphson_nonlocal_strategy.hpp [hunk 1]
```diff
@@ -104,7 +104,7 @@ class PoromechanicsNewtonRaphsonNonlocalStrategy : public PoromechanicsNewtonRap
                     mpNonlocalDamageUtility = new NonlocalDamage3DUtilities();
```
Comment (loumalouomega) [inline applications/PoromechanicsApplication/custom_strategies/strategies/poromechanics_newton_raphson_nonlocal_strategy.hpp:104]: are you checking 2D but setting 3D

File: applications/PoromechanicsApplication/custom_strategies/strategies/poromechanics_ramm_arc_length_strategy.hpp [hunk 2]
```diff
@@ -451,10 +468,10 @@ class PoromechanicsRammArcLengthStrategy : public PoromechanicsNewtonRaphsonStra
     int Check() override
     {
         KRATOS_TRY
-        
+
         int ierr = MotherType::Check();
         if(ierr != 0) return ierr;
-        
+
         if(ARC_LENGTH_LAMBDA.Key() == 0)
```
Comment (loumalouomega) [inline applications/PoromechanicsApplication/custom_strategies/strategies/poromechanics_ramm_arc_length_strategy.hpp]: Use the check macros

**Review by loumalouomega — CHANGES_REQUESTED** [review]
Minor comments

File: applications/PoromechanicsApplication/custom_strategies/strategies/poromechanics_newton_raphson_nonlocal_strategy.hpp (hunk 1 above)
Comment (ipouplana) [reply applications/PoromechanicsApplication/custom_strategies/strategies/poromechanics_newton_raphson_nonlocal_strategy.hpp:104]: You were looking at two non-consecutive lines

Comment (ipouplana) [reply applications/PoromechanicsApplication/custom_strategies/strategies/poromechanics_newton_raphson_nonlocal_strategy.hpp:104]: If it’s 2D, then I set 2D utility. The 3D is in the else part

**Review by loumalouomega — DISMISSED** [review]
Ups, sorry

File: applications/PoromechanicsApplication/custom_strategies/strategies/poromechanics_ramm_arc_length_strategy.hpp (hunk 2 above)
Comment (ipouplana) [reply applications/PoromechanicsApplication/custom_strategies/strategies/poromechanics_ramm_arc_length_strategy.hpp]: Done ;)

**Review by loumalouomega — APPROVED** [review]

**Pull request merged**